src/pages/EditorPage.js

```diff
@@ -339,28 +365,166 @@ export default function EditorPage(props) {
       }}
     >
-      Save Widget
+      Publish draft as main version
     </CommitButton>
   );
 
-  const widgetPath = `${accountId}/${path?.type}/${path?.name}`;
-  const jpath = JSON.stringify(path);
+  const saveLocallyButton = (
+    <button
+      className="btn btn-success"
+      onClick={(e) => {
+        e.preventDefault();
+        const key = `social.near.editor.widget.${accountId}`;
+        let widgetObjStr = localStorage.getItem(key);
+        if (widgetObjStr === null) {
+          const data = isDraft ? {
+            [widgetName]: {
+              branch: {
+                draft: {
+                  "": code
+                }
+              }
+            },
+          } : {
+            [widgetName]: {
+              "": code,
+              metadata,
+            },
+          };
+          localStorage.setItem(key, JSON.stringify(data));
+        } else {
+          let widgetObj = JSON.parse(widgetObjStr);
+          if (isDraft) {
+            widgetObj[widgetName] = {
+              ...widgetObj[widgetName],
+              branch: {
+                draft: {
+                  "": code
+                }
+              }
+            }
+          } else {
+            widgetObj[widgetName] = {
+              ...widgetObj[widgetName],
+              "": code,
+              metadata
+            }
+          }
+          localStorage.setItem(key, JSON.stringify(widgetObj));
+        }
+      }}
+    >
+      Save {isDraft ? "Draft" : "Widget"} in Local Storage
+    </button>
+  );
+
+  const saveDraftButton = (
+    <button
+      className="btn btn-danger"
+      disabled={!widgetName}
+      onClick={(e) => {
+        e.preventDefault();
+        setShowSaveDraftModal(true);
+      }}
+    >
+      Save Draft on chain
+    </button>
+  );
+
+  const publishButton = (
+    <CommitButton
+      className="btn btn-danger"
+      disabled={!widgetName}
+      near={near}
+      data={{
+        widget: {
+          [widgetName]: {
+            "": code,
+            metadata,
+          },
+        },
+      }}
+    >
+      Publish Widget on chain
+    </CommitButton>
+  );
+
+  const renderPreviewButton = (
+    <button
+      className="btn btn-success"
+      onClick={() => {
+        setRenderCode(code);
+        if (layout === Layout.Tabs) {
+          setTab(Tab.Widget);
+        }
+      }}
+    >
+      Render preview
+    </button>
+  );
+
+  const openCreateButton = (
+    <button className="btn btn-primary" onClick={() => setShowOpenModal(true)}>
+      Open / Create
+    </button>
+  );
+
+  const renameButton = (
+    <button
+      className="btn btn-secondary"
+      onClick={() => {
+        setShowRenameModal(true);
+      }}
+    >
+      Rename {path?.type}
+    </button>
+  );
+
+  const openInNewTabButton = (
+    <a
+      className="btn btn-outline-primary"
+      href={`#/${widgetPath}`}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      Open Widget page in new tab
```

Comment:
Besides the label, Is there any difference between this and the original `Open Component in a new tab` button?

Reply:
I don't think so but let's wait for @marcinbodnar to confirm. Should we rename "widget" to "component" in all relevant places in the editor? What should be the naming convention here?

Reply:
Yes lets use component for now. Before this is approved for merge, please share this flow and preview with Corwin to see if he suggests any changes to the UX.

Reply:
@charleslavon yes, it's just a different label